src/Chapter2.hs

```diff
@@ -841,7 +849,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: a -> [a] -> [a]
+rotate n l
+  | n < 0 || null l = []
+  | otherwise = drop n (take (length 1+n) (cycle l))
```

Comment:
That's a correct solution! Though, it does a little bit more than work when the given number is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face: